### PR TITLE
Extend IR hasing to 128 bits. Fix issue where same graph is fed by di…

### PR DESCRIPTION
**Pull request by dlibenzi**
…fferent parameter sequences.

### CODE REVIEW

File: third_party/xla_client/util.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+size_t StdHashCombine(uintmax_t a, uintmax_t b) {
+  return a ^
+         (b * 0x27d4eb2f165667c5 + 0x9e3779b97f4a7c15 + (a << 6) + (a >> 2));
+}
+
+hash_t HashCombine(const hash_t& a, const hash_t& b) {
+  static const hash_t kb = absl::MakeUint128(101, 0x27d4eb2f165667c5);
+  return a ^ (b * kb + 0x9e3779b97f4a7c15 + (a << 6) + (a >> 2));
+}
```
Comment (JackCaoG) [inline third_party/xla_client/util.cc]: Question, how do you choose when to use `stdHashCombine`(32 bits) and when to use `HashCombine`(128 bits)?

Comment (dlibenzi) [reply third_party/xla_client/util.cc]: Yeah, the multiplication and left shift moves entropy in the upper bits automatically.

File: test/test_operations.py [hunk 2]
```diff
@@ -1413,6 +1413,23 @@ def test_fn(t, r):
 
     self.runAtenTest([torch.rand(2, 4), torch.zeros(2, 1)], test_fn)
 
+  def test_spooky_ailing(self):
```
Comment (ailzhang) [inline test/test_operations.py]: lol I can see my name all over the tests now :D 
btw this is essentially `test_pow` in `torch_test_meta.py`.

Comment (dlibenzi) [reply test/test_operations.py]: I like that name 😄 Can I keep it? Please ...

**Comment by dlibenzi** [comment]
So the issue was with the parameter dedup.
The computation was exactly the same, but the parameters were deduped in a different order.
We need parameter dedup since we can otherwise hit the limit the TPU currently has with the maximum number of XLA computation parameters.
If we disabled dedup, we had no issues.
This is how I spotted the issue.
Setting `XLA_DEVDATA_CACHE_SIZE=0` made sure every time we build device data for a constant, we create a new one.
That effectively removed the deduped entries (all were unique), and make your spooky test to succeed.

**Pull request merged**